planner/core/logical_plan_builder.go

```diff
@@ -1941,15 +1947,15 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint) {
 	var preferAggType uint
 	for _, hint := range hints {
 		switch hint.HintName.L {
-		case TiDBMergeJoin:
+		case TiDBMergeJoin, HintSMJ:
```

Comment:
It seems `TiDBMergeJoin` is duplicated. Shall we consider removing it? Also take `TiDBIndexNestedLoopJoin` and  `TiDBHashJoin` into consideration.